src/function/audio.rs

```diff
@@ -0,0 +1,296 @@
+//! USB Audio Class 2 (UAC2) function.
+//!
+//! The Linux kernel configuration option `CONFIG_USB_CONFIGFS_F_UAC2` must be enabled.
+//!
+//! # Example
+//!
+//! ```no_run
+//! use usb_gadget::function::audio::Uac2;
+//! use usb_gadget::{default_udc, Class, Config, Gadget, Id, Strings};
+//!
+//! // capture: 8 ch, 48000 Hz, 24 bit, playback: 2 ch, 48000 Hz, 16 bit
+//! let (audio, func) = Uac2::new((0b1111_1111, 48000, 24 / 8), (0b11, 48000, 16 / 8)).build();
+//!
+//! let udc = default_udc().expect("cannot get UDC");
+//! let reg =
+//!     // USB device descriptor base class 0, 0, 0: use Interface Descriptors
+//!     // Linux Foundation VID Gadget PID
+//!     Gadget::new(Class::new(0, 0, 0), Id::new(0x1d6b, 0x0104), Strings::new("Clippy Manufacturer", "Rust UAC2", "RUST0123456"))
+//!         .with_config(Config::new("Audio Config 1").with_function(func))
+//!         .bind(&udc)
+//!         .expect("cannot bind to UDC");
+//!
+//! println!(
+//!     "UAC2 audio {} at {} to {} status {:?}",
+//!     reg.name().to_string_lossy(),
+//!     reg.path().display(),
+//!     udc.name().to_string_lossy(),
+//!     audio.status()
+//! );
+//! ```
+
+use std::{ffi::OsString, io::Result};
+
+use super::{
+    util::{FunctionDir, Status},
+    Function, Handle,
+};
+
+/// Audio device configuration.
+///
+/// Fields are optional and will be set to f_uac2 default values if not specified, see drivers/usb/gadget/function/u_uac2.h. Not all fields are supported by all kernels; permission denied errors may occur if unsupported fields are set.
```

Comment:
> Not all fields are supported by all kernels; permission denied errors may occur if unsupported fields are set.

If you wish, you could add `is_xxx_supported() -> bool` functions that use [`crate::linux_version`](https://github.com/surban/usb-gadget/blob/2c2a45673eb68f25693dd75d60315cd6d21648a1/src/lib.rs#L133) to check whether a setting is supported.

You could also define a macro that can be used for defining such functions.

However, this can also be done at a later time in a separate PR. Probably other gadgets suffer from the same problem, but nobody has noticed yet.